MCP - configure name and version for the published version

## Problem

There is an attempt to read the missing package.json in the published version :

```bash
npx -y @ignfab/geocontext
```

=> unnamed-mcp-server@0.0.0

```
[2025-10-31T21:37:13.301Z] [WARN] Could not read package.json from project root: Error: ENOENT: no such file or directory, open '/mnt/data/workspaces/workspace_geocontext/demo-geocontext/package.json'
[2025-10-31T21:37:13.301Z] [ERROR] Couldn't find project name in package json
[2025-10-31T21:37:13.301Z] [WARN] Could not read package.json from project root: Error: ENOENT: no such file or directory, open '/mnt/data/workspaces/workspace_geocontext/demo-geocontext/package.json'
[2025-10-31T21:37:13.301Z] [INFO] Initializing MCP Server: unnamed-mcp-server@0.0.0
```

## Resource

See https://mcp-framework.com/docs/server-configuration/#server-name-and-version :

```js
const server = new MCPServer({
  name: "my-mcp-server",     // Default: package.json name or "unnamed-mcp-server"
  version: "1.0.0"          // Default: package.json version or "0.0.0"
});
```

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@ignfab/geocontext",
-  "version": "0.8.0",
+  "version": "0.8.1",
   "description": "An experimental MCP server providing spatial context for LLM.",
   "type": "module",
   "bin": {

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,19 +1,24 @@
 import { MCPServer } from "mcp-framework";
 
-import { dirname } from "path";
+import { dirname, join } from "path";
 import { fileURLToPath } from "url";
+import { readFileSync } from "fs";
 
 // Get the directory of the current module (dist directory)
 const __dirname = dirname(fileURLToPath(import.meta.url));
 
+// Get version from package.json
+const pkgMetadata = JSON.parse(readFileSync(join(__dirname, '../package.json'), 'utf-8'));
+const VERSION = pkgMetadata.version;
+
 /**
  * Available transports
  */
-const TRANSPORTS : Record<string, any> = {
-  'stdio' : {
+const TRANSPORTS: Record<string, any> = {
+  'stdio': {
     type: "stdio"
   },
-  'http' : {
+  'http': {
     type: "http-stream",
     options: {
       port: 3000,
@@ -28,12 +33,14 @@ const TRANSPORTS : Record<string, any> = {
 async function main() {
   // get transport type from environment variable
   const TRANSPORT_TYPE = process.env.TRANSPORT_TYPE || "stdio";
-  if ( TRANSPORT_TYPE !== "stdio" && TRANSPORT_TYPE !== "http" ){
+  if (TRANSPORT_TYPE !== "stdio" && TRANSPORT_TYPE !== "http") {
     throw new Error(`Invalid transport type: ${TRANSPORT_TYPE}`);
   }
 
   // start MCP server
   const mcpServer = new MCPServer({
+    name: 'geocontext',
+    version: VERSION,
     basePath: __dirname,
     transport: TRANSPORTS[TRANSPORT_TYPE],
   });

